src/components/TimeBasedLineChart.js

```diff
@@ -1,32 +1,231 @@
+    position: 'relative',
+    boxSizing: 'content-box',
+    display: 'inline-block'
+  },
+
+  // Chart
+  breakPointLabel: {
+    fill: StyleConstants.Colors.ASH,
+    'font-family': StyleConstants.Fonts.REGULAR,
+    'font-size': StyleConstants.FontSizes.SMALL,
+    stroke: 'none'
+  },
+  breakPointLine: {
+    fill: 'none',
+    stroke: StyleConstants.Colors.FOG,
+    'stroke-width': 1
+  },
+  circle: {
+    fill: StyleConstants.Colors.WHITE,
+    'stroke-width': 2
+  },
+  dateTooltip: {
+    fill: StyleConstants.Colors.CHARCOAL,
+    stroke: 'none'
+  },
+  dateTooltipText: {
+    fill: StyleConstants.Colors.FOG,
+    stroke: 'none',
+    'font-family': StyleConstants.Fonts.REGULAR,
+    'font-size': StyleConstants.FontSizes.MEDIUM
+  },
+  domain: {
+    opacity: 0
+  },
+  text: {
+    'font-family': StyleConstants.Fonts.REGULAR,
+    'font-size': StyleConstants.FontSizes.MEDIUM,
+    stroke: 'none'
+  },
+  verticalLine: {
+    fill: 'none',
+    stroke: StyleConstants.Colors.ASH,
+    'stroke-width': 1
+  },
+  xAxisLabel: {
+    fill: StyleConstants.Colors.ASH,
+    stroke: 'none'
+  },
+  yAxisLabel: {
+    stroke: 'none',
+    'text-anchor': 'end'
+  },
+
+  // Hovered Data Point
+  hoveredDataPointDetail: {
+    boxSizing: 'border-box',
+    display: 'inline-block',
+    float: 'left'
+  },
+  hoveredDataPointDetails: {
+    padding: 20,
+    width: '100%'
+  },
+  hoveredDataPointLabel: {
+    boxSizing: 'border-box',
+    color: StyleConstants.Colors.CHARCOAL,
+    display: 'inline-block',
+    fontFamily: StyleConstants.Fonts.REGULAR,
+    fontSize: StyleConstants.FontSizes.MEDIUM,
+    paddingRight: 5,
+    textAlign: 'right'
+  },
+  hoveredDataPointValue: {
+    boxSizing: 'border-box',
+    color: StyleConstants.Colors.CHARCOAL,
+    display: 'inline-block',
+    fontFamily: StyleConstants.Fonts.SEMIBOLD,
+    fontSize: StyleConstants.FontSizes.MEDIUM,
+    textAlign: 'left',
+    width: 90
+  },
+
+  // Zero State
+  zeroState: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)'
+  }
+};
 
+const HoveredDataPointGroup = React.createClass({
+  propTypes: {
+    adjustedHeight: React.PropTypes.number.isRequired,
+    hoveredDataPoint: React.PropTypes.object.isRequired,
+    rangeType: React.PropTypes.string.isRequired,
+    translation: React.PropTypes.string,
+    xScaleValueFunction: React.PropTypes.func.isRequired,
+    yScaleValueFunction: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      translation: 'translate(0,0)'
+    };
+  },
+
+  render () {
+    return (
+      <g className='hover-state' ref='hoverState'>
+        <g className='hover-state-line' ref='hoverStateLine' transform={this.props.translation}>
+          <line
+            className='hovered-data-point-line'
+            x1={this.props.xScaleValueFunction(this.props.hoveredDataPoint.x)}
+            x2={this.props.xScaleValueFunction(this.props.hoveredDataPoint.x)}
```

Comment:
Just an idea I'm throwing out here. Instead of repeating `this.props.something`, I wonder if you could grab these props at the top:
```javascript
{ hoverDataPoint, rangeType } = this.props;
```
And then use these `{xScaleValueFunction(hoveredDataPoint.x)}`
`{moment.unix(hoveredDataPoint.x).format(rangeType === 'day' ? 'MMM DD' : 'MMM')}`